src/marqo/tensor_search/tensor_search.py

```diff
@@ -452,8 +452,12 @@ def add_documents(config: Config, index_name: str, docs: List[dict], auto_refres
                     f"images for {batch_size} docs using {image_download_thread_count} threads ")
 
     if update_mode == 'replace' and use_existing_tensors:
-        # Get existing documents
-        doc_ids = [doc["_id"] for doc in docs if "_id" in doc]
+        doc_ids = []
+
+        # Iterate through the list in reverse, only latest doc with dupe id gets added.
+        for i in range(len(docs)-1, -1, -1):
+            if ("_id" in docs[i]) and (docs[i]["_id"] not in doc_ids):
+                doc_ids.append(docs[i]["_id"])
```

Comment:
what if there is a difference between the docs? This may be possible because it this operation isn't atomic. For example:
```
[{"_id": "1", "title":"Hello"}, 
{"_id": "1", "title":"World"}, 
{"_id": "1", "__chunks":[{"__title_vector": [0.1, 0.2]}]}
{"_id": "1", "__chunks":[{"__title_vector": [2.1, 0.3]}]}]
```

Reply:
In this case, only the last item would be considered `{"_id": "1", "__chunks":[{"__title_vector": [2.1, 0.3]}]}`.